### PR TITLE
run integration tests against true sqlite3 reference implementation locally

### PR DESCRIPTION
**Pull request by ethanniser**
- run integration tests against actual sqlite3
 - this makes sure that are tests are actually correct to spec

we might want to check the `sqlite3` version as well

### CODE REVIEW

**Comment by jussisaurio** [comment]
The tests are already run against a pinned version of sqlite in CI:

https://github.com/penberg/limbo/blob/main/.github/workflows/rust.yml#L55-L68

However not against adding a new makefile command to do this locally. Was that the intention?

**Comment by ethanniser** [comment]
Didn't realize it was in ci

But yes, I think it's still quite helpful when writing the tests locally

File: Makefile [hunk 1]
```diff
@@ -8,6 +8,9 @@ SQLITE_EXEC ?= ./target/debug/limbo
 # Static library to use for SQLite C API compatibility tests.
 SQLITE_LIB ?= ./target/debug/liblimbo_sqlite3.a
 
+# Reference implementation to check against
+ACTUAL_SQLITE_EXEC ?= sqlite3
```
Comment (penberg) [inline Makefile:12]: If we are doing this, then we might as well have `SQLITE_EXEC` and `LIMBO_EXEC`.

**Review by penberg — CHANGES_REQUESTED** [review]

**Comment by jussisaurio** [comment]
BTW @ethanniser if you're wondering about the `string_agg` error in CI, it's because the test runner has a default `sqlite3` in `PATH` whose version doesn't support `string_agg` yet. The github actions YAML file has a comment about that.

**Comment by penberg** [comment]
@ethanniser I did https://github.com/penberg/limbo/pull/226 but I guess it kind of overlaps with this pull request. So please just tweak the Github config to run the right make target that does _not_ build Limbo.

**Comment by ethanniser** [comment]
actually on second thought I think its fine how it is- seems fine to catch it in ci and im not too sure how to specify the version available in the make file

you can just merge #226 and I will close this unless you think this is strongly worth adding

**Comment by penberg** [comment]
I merged https://github.com/penberg/limbo/pull/226, which kind of obsoletes this. I am totally fine with adding a `test-sqlite` and `test-limbo` makefile targets, which use `EXEC_SQLITE` and `EXEC_LIMBO`, respectively, if someone feels like doing that to simplify the local development flow. However, I am closing this PR for now

**Pull request closed without merging**